tools/packaging/catalog/catalog-remote.js

```diff
@@ -435,6 +439,41 @@ Object.assign(Table.prototype, {
     self._deleteQuery = "DELETE FROM " + self.name + " WHERE _id=?";
   },
 
+  // Compress a JSON string for storage
+  _compressString: function(str) {
+    var self = this;
+    if (!self._compressContent) {
+      return str;
+    }
+    
+    try {
+      var compressed = zlib.gzipSync(Buffer.from(str, 'utf8'));
```

Comment:
Since we are using a sync version, I would like to have a benchmark in the big/small machine, and for big/small payloads to know how it impacts the I/O since it's a blocking code